langlib/langlib-test/src/test/resources/test-src/type-param/type_param_narrowing_for_union_return.bal

```diff
@@ -0,0 +1,48 @@
+function testSimpleUnionReturnParameterNarrowing() {
+    int[]|float[] arr = <int[]>[1, 2];
+    [int, (int|float)][] y = arr.enumerate();
+    assertTrue(true);
```

Comment:
While these are not tests for the langlib methods themselves (`enumerate`, `get`, etc.) IMO it'll still be useful to assert something from the result to verify correct runtime behaviour.

For example here we can assert
```ballerina
assertTrue(y[0][1] is int);
assertEquals(1, y[0][1]);
assertTrue(y[1][1] is int);
assertEquals(2, y[1][1]);
```

Shall we replace the `assertTrue(true)`s with proper assertions for type and value?